sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1464,12 +1465,14 @@ case class ArrayContains(left: Expression, right: Expression)
     nullSafeCodeGen(ctx, ev, (arr, value) => {
       val i = ctx.freshName("i")
       val getValue = CodeGenerator.getValue(arr, right.dataType, i)
+      val setIsNullCode = if (nullable) s"${ev.isNull} = true;" else ""
+      val unsetIsNullCode = if (nullable) s"${ev.isNull} = false;" else ""
       s"""
       for (int $i = 0; $i < $arr.numElements(); $i ++) {
         if ($arr.isNullAt($i)) {
-          ${ev.isNull} = true;
+          ${setIsNullCode}
```

Comment:
nit: `${setIsNullCode}` -> `$setIsNullCode`
btw, you found some bugs when excluding the `ConvertToLocalRelation` rule in tests, right?

Reply:
Please open separate JIRAs and PRs for the codegen fixes.

Reply:
@gatorsmile OK.. Sean.